Breakout/src/game_clock.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <vector>
+#include <functional>
+#include <SFML/System/Clock.hpp>
+
+class GameClock
+{
+public:
+	/* Note: Clock starts at construction*/
+	explicit GameClock(float);
+
+	/* Starts clock at 0, run when you want to begin a new instance of the clock.*/
+	void initiateClock();
+
+	/* Subscribe update methods to clock to recieve a callback when clock ticks */
+	void subscribe(std::function<void(float)>);
```

Comment:
```suggestion
	void subscribe(std::function<void(float)> callbackSub);
```

Reply:
Can you clarify something for me? Is it fine to give the clock a callback system, allowing clients to subscribe to clock ticks? You are saying its best to not include the game update method in that callback system however, as control should remain with the client for that specific aspect of the game.

Thanks for taking the time btw

Reply:
I don't have a problem with such a callback system per se. However if you look at it from a design perspective, the clock should be a self contained unit and stay completely isolated from anything else in the application.
If you really want to have a callback system, I would recommend creating some kind of manager where objects can subscribe to specific events. That way you have logically separated units doing exactly what their name supposes them to do, a simple clock and a state updater or however you want to call it.

Reply:
Ok. Makes sense. Thank you. I'll leave the functionality there for now but remove its actual usage.